base/math.jl

```diff
@@ -784,7 +784,7 @@ function ldexp(x::T, e::Integer) where T<:IEEEFloat
     xu = reinterpret(Unsigned, x)
     xs = xu & ~sign_mask(T)
     xs >= exponent_mask(T) && return x # NaN or Inf
-    k = Int(xs >> significand_bits(T))
+    k = (xs >> significand_bits(T)) % Int
```

Comment:
What's the difference between these?

Reply:
`Int` throws if the sign bit is set and inference does not have any range modeling to know that that is impossible.

Reply:
Ah. That's somewhat unfortunate, but this is a good fix then.